net: lib: dhcpv4: added router id in response options

munir-zin opened the pull request:
When running dhcp, it was not assigning gateway/router address to the client devices. It is fixed in this PR.

rlubos commented on subsys/net/lib/dhcpv4/dhcpv4_server.c:
```diff
@@ -379,6 +380,22 @@ static uint8_t *dhcpv4_encode_subnet_mask_option(uint8_t *buf, size_t *buflen,
 }
 
+static uint8_t *dhcpv4_encode_router_option(uint8_t *buf, size_t *buflen,
+						 struct in_addr *router)
```
Please align with the opening bracket.

rlubos commented on subsys/net/lib/dhcpv4/dhcpv4_server.c:
```diff
@@ -467,9 +484,11 @@ static uint8_t *dhcpv4_encode_requested_params(
 		case DHCPV4_OPTIONS_SUBNET_MASK:
 			buf = dhcpv4_encode_subnet_mask_option(
 				buf, buflen, &ctx->netmask);
-			if (buf == NULL) {
-				goto out;
-			}
```
Please don't remove this check, we need to break the loop in case option encoding fails.

rlubos commented on subsys/net/lib/dhcpv4/dhcpv4_server.c:
```diff
@@ -467,9 +484,11 @@ static uint8_t *dhcpv4_encode_requested_params(
+		case DHCPV4_OPTIONS_ROUTER:
+			buf = dhcpv4_encode_router_option(
+				buf, buflen, &ctx->iface->config.ip.ipv4->gw);
 			break;
```
And a similar check should be added here.

rlubos reviewed (commented):
Generally looks good, just some minor comments. Thanks

rlubos commented on subsys/net/lib/dhcpv4/dhcpv4_server.c:
```diff
@@ -379,6 +380,22 @@ static uint8_t *dhcpv4_encode_subnet_mask_option(uint8_t *buf, size_t *buflen,
 }
 
+static uint8_t *dhcpv4_encode_router_option(uint8_t *buf, size_t *buflen,
+						                    struct in_addr *router)
```
The alignment is still off, note that Zephyr uses 8 spaces/tab.

munir-zin replied:
When using all tabs, it may not align completely. Can I use spaces for last tab?

rlubos commented:
Please squash the commits.

rlubos commented:
So the diff looks good for me, but there's some mess with commits - we just want a single commit with the change. Please use interactive rebase and squash them (also no merge commts please, use rebase instead).

rlubos commented:
> > So the diff looks good for me, but there's some mess with commits - we just want a single commit with the change. Please use interactive rebase and squash them (also no merge commts please, use rebase instead).
> 
> I followed following link to squash it. The last commit 5 which shows merge is a result of squash. Can you please guide me how I do it correctly as we follow different methods of PR and this rebasing and squash I am doing for the first time. https://forum.freecodecamp.org/t/how-to-squash-multiple-commits-into-one-with-git-squash/13231

The guidelines in the link seems fine, I'm not sure we'v ended up like this. The exact steps I'd take in this case (while on branch):
1. `git rebase -i HEAD~5` - this starts interactive rebase
2. leave the topmost commit as is, the remaining ones below, replace `pick` with `squash` (or simply `s`).
3. `git log` - you should only see one new commit now
4. `git push -f `

maass-hamburg commented:
I think you now have squashed all into an unrelated commit. You should only have squashed the ones after your first one.

munir-zin commented:
> I think you now have squashed all into an unrelated commit. You should only have squashed the ones after your first one.

I added only 5 commits of mine, I am not sure how these new commits are added.

ycsin commented:
> > I think you now have squashed all into an unrelated commit. You should only have squashed the ones after your first one.
> 
> I added only 5 commits of mine, I am not sure how these new commits are added.

You need to rebase your changes on top of Zephyr main branch, instead of pulling zephyr main branch into your PR branch

munir-zin committed:
net: lib: dhcpv4: added router id in response options

When running dhcp, it was not assigning gateway/router
address to the client devices. It is fixed in this PR.

Signed-off-by: Muhammad Munir <muhammad.munir@zintechnologies.com>

munir-zin commented:
> > > I think you now have squashed all into an unrelated commit. You should only have squashed the ones after your first one.
> > 
> > 
> > I added only 5 commits of mine, I am not sure how these new commits are added.
> 
> You need to rebase your changes on top of Zephyr main branch, instead of pulling zephyr main branch into your PR branch
> 
> To fix this, you will have to:
> 
> * hard reset your local PR branch to the tip of upstream main branch (this will remove any local changes you have)
> * apply your changes to the local PR branch, and commit (do not push yet)
> * force push your local branch to your remote
> 
> This happens a lot to new contributors, don't fret, it took me 3 PRs to make it right last time.

Please verify now. I hope it's good this time.

rlubos commented:
There're still some compliance nits:
```
2: B4 Second line is not empty: "When running dhcp, it was not assigning gateway/router address to the client devices. It is fixed in this PR."
2: UC4 Commit message body line exceeds max length (109>75): "When running dhcp, it was not assigning gateway/router address to the client devices. It is fixed in this PR."
Error: See https://docs.zephyrproject.org/latest/contribute/guidelines.html#coding-style for more details.

 TRAILING_WHITESPACE: trailing whitespace
File:subsys/net/lib/dhcpv4/dhcpv4_server.c
Line:498
```
Please fix the trailing whitespace, update the commit (`git commit --amend`), and fix the commit message along the way.

ycsin commented on subsys/net/lib/dhcpv4/dhcpv4_server.c:
```diff
@@ -472,6 +489,13 @@ static uint8_t *dhcpv4_encode_requested_params(
+			if (buf == NULL) {
+				goto out;
+			}
+			break;	
```
If you are using vscode, have a look at [editorconfig](https://marketplace.visualstudio.com/items?itemName=EditorConfig.EditorConfig) & [checkpatch](https://github.com/ycsin/zephyrproject-extension-pack/tree/pr/update_readme_0?tab=readme-ov-file#checkpatch) to minimize this kind of stuff

rlubos commented:
One more to go:
>  2: B4 Second line is not empty: "When running dhcp, it was not assigning gateway/router"

munir-zin commented:
> One more to go:
> 
> > 2: B4 Second line is not empty: "When running dhcp, it was not assigning gateway/router"

I could not get it. I tried to reduce the length of characters per line. Do I need to add only one line in commit's message?

maass-hamburg commented:
> > One more to go:
> > > 2: B4 Second line is not empty: "When running dhcp, it was not assigning gateway/router"
> 
> I could not get it. I tried to reduce the length of characters per line. Do I need to add only one line in commit's message?

One empty line is required between the first line and the following lines.

munir-zin commented:
Thanks for the approvals. I am not authorized to merge it.
